rust/lance-core/src/datatypes/field.rs

```diff
@@ -516,6 +516,23 @@ impl Field {
             .unwrap_or(false)
     }
 
+    /// If the field is a blob, return a new field with the same name and id
```

Comment:
This comment says "return a new field" but it is modifying the existing field.  Also, if it is going to modify the existing field, does it need to return self?  That introduces an awkward `let _ = ...` below.